src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py

```diff
@@ -6,6 +6,49 @@
 from ._util import ParametersContext, patch_arg_make_required
 
 
+with ParametersContext(command='sql db') as c:
+    c.register_alias('database_name', ('--name', '-n'))
+
+with ParametersContext(command='sql db data-warehouse') as c:
+    c.register_alias('database_name', ('--database-name',))
```

Comment:
`data-warehouse` group has only action commands like `pause` & `resume` and they operate on --database-name, --server-name & --resource-group but as the last noun in the command is data-warehouse I did de-alias the --database-name as per the convention.